Remove extra / from Logo url when using S3 Bucket Storage

misilot opened the pull request:
By having the extra / and using an external storage service makes the logo not load due to the extra /.
This was tested with S3 and local storage and both continued to work afterwards

# Description

By having the extra ```/``` in the code URL's are generated like https://bucket.s3.amazonaws.com/<root>//setting-logo-zSdoVnTFSz.jpg and S3 will not serve it. By removing the extra ```/``` the URL is generated as https://bucket.s3.amazonaws.com/<root>/setting-logo-zSdoVnTFSz.jpg and displays as expected

## Type of change

Please delete options that are not relevant.

- [X] Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Test A
- [ ] Enable storage on S3
- [ ] Upload Logo
- [ ] See logo does not display
- [ ] Apply Patch
- [ ] php artisan config:cache
- [ ] php artisan cache:clear
- [ ] Refresh Page, see logo

Test B (Local storage)
- [ ] Upload Logo
- [ ] See logo there
- [ ] Apply Patch
- [ ] php artisan config:cache
- [ ] php artisan cache:clear
- [ ] Refresh page, logo is still there

# Checklist:

- [X] I have read the Contributing documentation available here: https://snipe-it.readme.io/docs/contributing-overview
- [X] I have formatted this PR according to the project guidelines: https://snipe-it.readme.io/docs/contributing-overview#pull-request-guidelines
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works

snipe commented:
I'm not sure why this would be needed though, unless your APP_URL has a trailing slash, which is shouldn't.

snipe commented:
(Also please make PRs against develop, not master, per the contributor docs - thanks!)

misilot committed:
Fixes Site Logo Not loading when using S3 Bucket for public storage

When using an S3 bucket for the public storage and the "url('/')" adds
an extra / to the URL which S3 does not strip out, therefore does not
match an item in the bucket.

Tested with before/after with Local storage and S3 Storage

misilot commented:
Well my ```APP_URL``` is set to null. I did have a trailing slash in```PUBLIC_AWS_BUCKET_ROOT```, but adding ```APP_URL``` and removing the trailing slash did not resolve the logo from the S3 bucket loading (due to the extra /) until I removed the / from the ('/') line.

snipe commented:
Your APP_URL should not be set to null tho, per [the configuration instructions](https://snipe-it.readme.io/docs/configuration#required-basic-app-settings).

misilot commented:
I did update ```APP_URL``` per the instructions when you had mentioned it earlier.

snipe commented:
So then this should not be necessary then? I'm just not really clear what problem this is solving. If this was necessary, the logo would never have loaded

misilot commented:
I think it still needs to be.

I will take screenshots and copy the links generated for the images and share them with the following scenarios. As well as a copy of my .env file sanitized.  (APP_* and *S3* settings)

1. Local Storage before patch
2. Local storage after patch
3. S3 Storage before patch
4. S3 Storage after patch

misilot commented:
Sorry, I should have pictures after lunch. I ran into CSP issues and the public url not being added to the CSP policy #9541  that looks to be always enabled (unless debug mode is on), I will submit a PR for that.

misilot commented:
This is similar to the logo preview (see: https://github.com/snipe/snipe-it/blob/master/resources/views/partials/forms/edit/uploadLogo.blade.php#L19)

### Pre patch local:
```html
<a class="logo navbar-brand no-hover" href="https://server">
    <img class="navbar-brand-img" src="https://server/uploads/setting-logo-LQZ4mJQW17.jpg" alt="Asset Management Test System logo">
    Asset Management Test System
</a>
```

<img width="824" alt="image" src="https://user-images.githubusercontent.com/1446856/117190686-5e67c480-ada5-11eb-9fcb-a33d2622f0c5.png">

### Post patch local:
```html
<a class="logo navbar-brand no-hover" href="https://server">
    <img class="navbar-brand-img" src="https://server/uploads/setting-logo-LQZ4mJQW17.jpg" alt="Asset Management Test System logo">
    Asset Management Test System
</a>
```

<img width="824" alt="image" src="https://user-images.githubusercontent.com/1446856/117190829-80f9dd80-ada5-11eb-98a7-226ced57ce4d.png">


### Pre patch s3:
```html
<a class="logo navbar-brand no-hover" href="https://server">
    <img class="navbar-brand-img" src="https://snipeit-inventory-storage.s3.amazonaws.com/s3root//setting-logo-SJP2Tm9P2x.jpg" alt="Asset Management Test System logo">
    Asset Management Test System
</a>
```

<img width="824" alt="image" src="https://user-images.githubusercontent.com/1446856/117190512-23fe2780-ada5-11eb-8ed7-79e7d362a702.png">


### Post patch s3:
```html
<a class="logo navbar-brand no-hover" href="https://server">
    <img class="navbar-brand-img" src="https://snipeit-inventory-storage.s3.amazonaws.com/s3root/setting-logo-SJP2Tm9P2x.jpg" alt="Asset Management Test System logo">
    Asset Management Test System
</a>
```
<img width="824" alt="image" src="https://user-images.githubusercontent.com/1446856/117190525-282a4500-ada5-11eb-9e72-f4383cd58c8c.png">



### ENV Config
```
APP_ENV=production
APP_DEBUG=false
APP_KEY=base64:<removed>
APP_URL=https://server
#APP_URL=null
APP_TIMEZONE='America/Chicago'
APP_LOCALE=en
MAX_RESULTS=500
PRIVATE_FILESYSTEM_DISK=s3_private
PUBLIC_FILESYSTEM_DISK=s3_public

#PRIVATE_FILESYSTEM_DISK=local
#PUBLIC_FILESYSTEM_DISK=local_public

# --------------------------------------------
# OPTIONAL: PUBLIC S3 Settings
# --------------------------------------------
PUBLIC_AWS_SECRET_ACCESS_KEY=<key>
PUBLIC_AWS_ACCESS_KEY_ID=<id>
PUBLIC_AWS_DEFAULT_REGION=us-east-1
PUBLIC_AWS_BUCKET=snipeit-inventory-storage
PUBLIC_AWS_URL=https://snipeit-inventory-storage.s3.amazonaws.com
PUBLIC_AWS_BUCKET_ROOT=/s3root

# --------------------------------------------
# OPTIONAL: PRIVATE S3 Settings
# --------------------------------------------
PRIVATE_AWS_ACCESS_KEY_ID=<id>
PRIVATE_AWS_SECRET_ACCESS_KEY=<key>
PRIVATE_AWS_DEFAULT_REGION=us-east-1
PRIVATE_AWS_BUCKET=snipeit-inventory-storage
PRIVATE_AWS_URL=null
PRIVATE_AWS_BUCKET_ROOT=/s3root
```

<blockquote><img src="https://repository-images.githubusercontent.com/14346979/164a6580-7d79-11e9-865f-eae87593e11a" width="48" align="right"><div><img src="https://github.githubassets.com/favicons/favicon.svg" height="14"> GitHub</div><div><strong><a href="https://github.com/snipe/snipe-it">snipe/snipe-it</a></strong></div><div>A free open source IT asset/license management system - snipe/snipe-it</div></blockquote>

back-2-95 commented:
I can confirm this is an issue (v5.1.8) and I don't have trailing slash in APP_URL.

back-2-95 reviewed (approved):
Tested the changes with S3 and the logo appeared.

snipe commented:
If someone has Snipe-IT installed in a subdirectory (`https://foo.com/snipeit`), does this still work for local file storage?
